server/routes/creatingReports/whyWasUoFApplied.ts

```diff
@@ -5,52 +5,79 @@ import { paths } from '../../config/incident'
 
 const FORM = 'reasonsForUseOfForce'
 
+enum SubmitType {
+  SAVE_AND_CONTINUE = 'save-and-continue',
+  SAVE_AND_RETURN = 'save-and-return',
+}
+
 export default class WhyWasUoFAppliedRoutes {
   constructor(private readonly draftReportService: DraftReportService) {}
 
   public view() {
     return async (req: Request, res: Response): Promise<void> => {
       const { bookingId } = req.params
-      const { reasons } = await this.draftReportService.getSelectedReasonsForUoF(req.user.username, Number(bookingId))
+      const { isComplete, reasons } = await this.draftReportService.getUoFReasonState(
+        req.user.username,
+        Number(bookingId)
+      )
+
+      const selectedReasons = req.flash('clearingOutReasons')?.length ? [] : reasons
```

Comment:
if there are some persisted reasons and the user deselects them all and hits submit it will trigger the validation  message. 

in this scenario we don't want to present the previously persisted values as the user has explicitly removed all of them